src/api/0x/index.ts

```diff
@@ -0,0 +1,136 @@
+const EXCLUDED_SOURCES: ERC20BridgeSource[] = []
+const API_VERSION = 'v1'
+
+function get0xUrls(): Partial<Record<SupportedChainId, string>> {
+  // Support: Mainnet, Ropsten, Polygon, Binance Smart Chain
+  // See https://0x.org/docs/api#introduction
+  return {
+    [SupportedChainId.MAINNET]: 'https://api.0x.org/swap',
+  }
+}
+
+function _getMatchaChainId(chainId: SupportedChainId): SupportedChainId | null {
+  switch (chainId) {
+    // Support: Mainnet, Ropsten, Polygon, Binance Smart Chain
+    // See https://0x.org/docs/api#introduction
+    // but we only support mainnet of that list so..
+    case SupportedChainId.MAINNET:
+      return chainId
+
+    default:
+      // Unsupported network
+      return null
+  }
+}
+
+function _optionsToSearchUrl(options: MatchaOptions) {
+  const { excludedSources, affiliateAddress } = options
+
+  const excludedSourceString = excludedSources.length > 0 ? excludedSources.join(',') : ''
+  const optionsMap = Object.entries({ excludedSources: excludedSourceString, affiliateAddress }).map(
+    ([option, value]) => `${option}=${value}`
+  )
+  return optionsMap.join('&')
+}
+
+async function _handleQuoteResponse<T = unknown, P extends PriceQuoteParams = PriceQuoteParams>(
+  response: Response,
+  params?: P
+): Promise<T> {
+  if (!response.ok) {
+    const errorObj: ZeroXErrorResponse = await response.clone().json()
+
+    let priceError
+    if (errorObj?.code && errorObj?.reason) {
+      priceError = new ZeroXError(errorObj)
+    } else {
+      const errorMessage = await ZeroXError.getErrorFromStatusCode(response.clone())
+      priceError = new ZeroXError({ code: response.status, reason: errorMessage })
+    }
+
+    if (params) {
+      const { quoteToken, baseToken } = params
+      log.error(logPrefix, `Error querying fee from API - quoteToken: ${quoteToken}, baseToken: ${baseToken}`)
+    }
+
+    throw priceError
+  } else {
+    return response.json()
+  }
+}
+
+export class ZeroXApi extends BaseApi {
+  MATCHA_OPTIONS
+  MATCHA_OPTIONS_URL
+
+  constructor(
+    context: Context,
+    matchaOptions: MatchaOptions = { affiliateAddress: AFFILIATE_ADDRESS, excludedSources: EXCLUDED_SOURCES }
+  ) {
+    super({ context, name: '0x', apiVersion: API_VERSION, getApiUrl: get0xUrls })
+    this.MATCHA_OPTIONS = matchaOptions
+    this.MATCHA_OPTIONS_URL = _optionsToSearchUrl(matchaOptions)
+  }
+
+  async getQuote(params: PriceQuoteParams, options: Options = {}): Promise<MatchaPriceQuote | null> {
+    const { amount, baseToken, quoteToken, kind } = params
+    const { chainId: customChainId, isDevEnvironment = this.context.isDevEnvironment } = options
+    const chainId = customChainId || (await this.context.chainId)
+    // this is handled via an error on L51 via the fetch call
+    // but we can handle it here to control the flow better
+    const networkId = _getMatchaChainId(chainId)
+    if (networkId == null) {
+      log.debug('[0x API] - Network not supported')
```

Comment:
shouldn't we reuse the logprefix in every log statement?

Reply:
good catch